mvp_onboarding.client/src/components/CustomerTable.jsx

```diff
@@ -0,0 +1,354 @@
+import {
+  Table,
+  Button,
+  Icon,
+  Modal,
+  Form,
+  Input,
+  Pagination,
+  Dropdown
+} from "semantic-ui-react";
+
+const CustomerTable = () => {
+  const [customers, setCustomers] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [deleteOpen, setDeleteOpen] = useState(false);
+  const [newOpen, setNewOpen] = useState(false);
+  const [editOpen, setEditOpen] = useState(false);
+  const [selectedCustomer, setSelectedCustomer] = useState(null);
+  const [createIsDisabled, setCreateIsDisabled] = useState(true);
+  const [editIsDisabled, setEditIsDisabled] = useState(true);
+  const [selectedAddress, setSelectedAddress] = useState("");
+  const [selectedName, setSelectedName] = useState("");
+
+  const [sortColumn, setSortColumn] = useState('Name');
+  const [sortDirection, setSortDirection] = useState('asc');
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalCount, setTotalCount] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+  const pageSizeOptions = [
+    { key: 10, text: '10', value: 10 },
+    { key: 20, text: '20', value: 20 },
+    { key: 50, text: '50', value: 50 },
+  ];
+
+  const handleSort = (column, direction) => {
+    setSortColumn(column);
+    setSortDirection(direction);
+  }
+
+  useEffect(() => {
+    fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+  }, [currentPage, pageSize, sortColumn, sortDirection]);
+
+  useEffect(() => {
+    setCreateIsDisabled(invalidSelectedCustomer);
+    setEditIsDisabled(invalidSelectedCustomer);
+  }, [selectedAddress, selectedName]);
+
+  const fetchCustomers = async (currentPage, pageSize, sortColumn, sortDirection) => {
+    try {
+      const data = await getCustomers(currentPage, pageSize, sortColumn, sortDirection);
+      if (data?.items?.length > 0) {
+        setCustomers(data?.items);
+        setTotalCount(data?.totalCount);
+      }
+    } catch (error) {
+      console.error("Failed to fetch customers", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+  const handleDelete = async () => {
+    if (selectedCustomer) {
+      try {
+        await deleteCustomer(selectedCustomer?.id);
+        setCustomers(customers.filter((customer) => customer.id !== selectedCustomer?.id));
+        setDeleteOpen(false);
+      }
+      catch (error) {
+        console.error("Failed to delete customer", error);
+      }
+    }
+  };
+
+  const handleEditSubmit = async () => {
+    let newCustomer = {
+      id: selectedCustomer.id,
+      name: selectedName,
+      address: selectedAddress
+    };
+    if (newCustomer) {
+      try {
+        await updateCustomer(selectedCustomer.id, newCustomer);
+        fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+      }
+      catch (error) {
+        console.error("Failed to update customer", error);
+      }
+      setEditOpen(false);
+    }
+
```

Comment:
The same for this one, your **let** variable and condition should be inside **try**. Also you can use **finally** to update the state to false